From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Subject: [PATCH] Adapt S10 runtime Wi-Fi policy to GZD7

Derived from the existing 80211AX and 6GHz patches. GZD7 forward input.
Hotspot SPF_6G remains unchanged here; its separate policy runs later.

--- a/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali
+++ b/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali
@@ -1528,7 +1528,7 @@
 .method public isSupported6Ghz()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
--- a/smali/com/samsung/android/server/wifi/SemWifiCoexManager.smali
+++ b/smali/com/samsung/android/server/wifi/SemWifiCoexManager.smali
@@ -2891,38 +2891,9 @@
 .end method
 
 .method public getWifiUwbCoexMode()Ljava/lang/String;
-    .locals 2
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/SemWifiCoexManager;->mWifiManagerProxy:Lcom/samsung/android/server/wifi/SemWifiManagerProxy;
-
-    invoke-virtual {v0}, Lcom/samsung/android/server/wifi/SemWifiManagerProxy;->isWifiEnabled()Z
-
-    move-result v0
-
-    if-eqz v0, :cond_0
-
-    iget-object p0, p0, Lcom/samsung/android/server/wifi/SemWifiCoexManager;->mWifiNative:Lcom/samsung/android/server/wifi/halclient/SemWifiNative;
-
-    const-string v0, "wlan0"
-
-    invoke-virtual {p0, v0}, Lcom/samsung/android/server/wifi/halclient/SemWifiNative;->getWifiUwbCoexMode(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p0
-
-    const-string v0, "getWifiUwbCoexMode - return str = "
-
-    invoke-static {v0, p0}, Lcom/samsung/android/server/wifi/SemApeServiceImplV1$1$$ExternalSyntheticOutline0;->m(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v0
-
-    const-string v1, "SemWifiCoexManager"
-
-    invoke-static {v1, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    return-object p0
-
-    :cond_0
-    const-string p0, ""
+    .locals 0
+
+    const/4 p0, 0x0
 
     return-object p0
 .end method
--- a/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
@@ -1919,7 +1919,7 @@
 .end method
 
 .method public static showSecProductFeature()Ljava/lang/String;
-    .locals 3
+    .locals 5
 
     new-instance v0, Ljava/lang/StringBuffer;
 
@@ -1951,7 +1951,41 @@
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
-    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=false\nSPF_11ax=true\nSPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=false\nSPF_MCF_AutoHotspot=false\nSPF_OWE=false\nSPF_Prio_Traffic=true\nSPF_WS_Lite=false\nSPF_wpa3=true\nSPF_JDM=in_house\n"
+    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=false\n"
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v3, "SPF_11ax="
+
+    invoke-direct {v1, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-static {}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getInstance()Lcom/samsung/android/server/wifi/SemWifiInjector;
+
+    move-result-object v3
+
+    invoke-virtual {v3}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getWifiApChipInfo()Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;
+
+    move-result-object v3
+
+    const/4 v4, 0x1
+
+    invoke-virtual {v3, v4}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->supportThisSoftApFeature(I)Z
+
+    move-result v3
+
+    invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
+
+    const-string v1, "SPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=false\nSPF_MCF_AutoHotspot=false\nSPF_OWE=false\nSPF_Prio_Traffic=true\nSPF_WS_Lite=false\nSPF_wpa3=true\nSPF_JDM=in_house\n"
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
--- a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo.smali
@@ -27,6 +27,16 @@
 
 
 # direct methods
+.method static bridge synthetic -$$Nest$misSupportedSoftAp11ax(Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;)Z
+    .locals 0
+
+    invoke-direct {p0}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->isSupportedSoftAp11ax()Z
+
+    move-result p0
+
+    return p0
+.end method
+
 .method static bridge synthetic -$$Nest$misSupportedSoftApPowerSave(Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;)Z
     .locals 0
 
--- a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
@@ -1737,11 +1737,11 @@
 .end method
 
 .method public supportThisSoftApFeature(I)Z
-    .locals 3
+    .locals 2
 
     const/4 v0, 0x1
 
-    if-eq p1, v0, :cond_2
+    if-eq p1, v0, :cond_1
 
     const/4 v1, 0x2
 
@@ -1749,11 +1749,9 @@
 
     const/4 v1, 0x3
 
-    const/4 v2, 0x0
-
     if-eq p1, v1, :cond_0
 
-    return v2
+    goto :goto_1
 
     :cond_0
     iget-boolean p1, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSupportPowerSave:Z
@@ -1762,22 +1760,36 @@
 
     iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
 
-    if-eqz p0, :cond_1
+    if-eqz p0, :cond_3
 
     invoke-static {p0}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->-$$Nest$misSupportedSoftApPowerSave(Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;)Z
 
     move-result p0
 
-    if-eqz p0, :cond_1
+    if-eqz p0, :cond_3
 
     goto :goto_0
 
     :cond_1
-    return v2
+    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
+
+    if-eqz p0, :cond_3
+
+    invoke-static {p0}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->-$$Nest$misSupportedSoftAp11ax(Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_3
 
     :cond_2
     :goto_0
     return v0
+
+    :cond_3
+    :goto_1
+    const/4 p0, 0x0
+
+    return p0
 .end method
 
 .method public supportWifiSharing()Z
--- a/smali/com/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner.smali
+++ b/smali/com/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner.smali
@@ -639,7 +639,7 @@
 
 # virtual methods
 .method public generate_wifi_scan(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;III)V
-    .locals 4
+    .locals 3
 
     const-string v0, "generate_wifi_scan,ssid"
 
@@ -685,7 +685,7 @@
 
     iput-object p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
 
-    const/16 p2, 0x8
+    const/4 p2, 0x2
 
     invoke-virtual {p1, p2}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
 
@@ -697,255 +697,238 @@
 
     iget p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
 
-    or-int/lit8 p1, p1, 0x4
+    or-int/2addr p1, p2
 
     iput p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
 
     :cond_0
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
-
-    const/4 p3, 0x2
-
-    invoke-virtual {p1, p3}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
+    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
+
+    const/4 p3, 0x1
+
+    or-int/2addr p1, p3
+
+    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
+
+    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
+
+    invoke-virtual {p1}, Landroid/net/wifi/WifiManager;->getConfiguredNetworks()Ljava/util/List;
 
     move-result-object p1
 
-    array-length p1, p1
-
-    if-eqz p1, :cond_1
-
-    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
-
-    or-int/2addr p1, p3
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
+    iget p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
+
+    const/4 p5, 0x3
+
+    if-ne p4, p2, :cond_1
+
+    iput p5, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
 
     :cond_1
-    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
-
-    const/4 p4, 0x1
-
-    or-int/2addr p1, p4
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->supportBand:I
-
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
-
-    invoke-virtual {p1}, Landroid/net/wifi/WifiManager;->getConfiguredNetworks()Ljava/util/List;
+    invoke-interface {p1}, Ljava/util/List;->iterator()Ljava/util/Iterator;
 
     move-result-object p1
 
-    iget p5, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
-
-    const/4 p6, 0x3
-
-    if-ne p5, p3, :cond_2
-
-    iput p6, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
+    const/4 p2, 0x0
+
+    move p4, p2
 
     :cond_2
-    invoke-interface {p1}, Ljava/util/List;->iterator()Ljava/util/Iterator;
-
-    move-result-object p1
-
-    const/4 p3, 0x0
-
-    move p5, p3
-
-    :cond_3
     :goto_0
     invoke-interface {p1}, Ljava/util/Iterator;->hasNext()Z
 
+    move-result p6
+
+    if-eqz p6, :cond_3
+
+    invoke-interface {p1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
+
+    move-result-object p6
+
+    check-cast p6, Landroid/net/wifi/WifiConfiguration;
+
+    iget-object v0, p6, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
+
+    if-eqz v0, :cond_2
+
+    invoke-direct {p0, v0}, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->removeDoubleQuotes(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v0
+
+    iget-object v2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-virtual {v0, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
     move-result v0
 
-    if-eqz v0, :cond_4
-
-    invoke-interface {p1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v0
-
-    check-cast v0, Landroid/net/wifi/WifiConfiguration;
-
-    iget-object v2, v0, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
-
-    if-eqz v2, :cond_3
-
-    invoke-direct {p0, v2}, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->removeDoubleQuotes(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v2
-
-    iget-object v3, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-virtual {v2, v3}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result v2
-
-    if-eqz v2, :cond_3
-
-    iget p5, v0, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
-
-    const-string v2, " existing profile removed"
-
-    invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
-
-    iget v0, v0, Landroid/net/wifi/WifiConfiguration;->networkId:I
-
-    invoke-virtual {v2, v0}, Landroid/net/wifi/WifiManager;->removeNetwork(I)Z
+    if-eqz v0, :cond_2
+
+    iget p4, p6, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
+
+    const-string v0, " existing profile removed"
+
+    invoke-static {v1, v0}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
+
+    iget p6, p6, Landroid/net/wifi/WifiConfiguration;->networkId:I
+
+    invoke-virtual {v0, p6}, Landroid/net/wifi/WifiManager;->removeNetwork(I)Z
 
     goto :goto_0
 
+    :cond_3
+    new-instance p1, Landroid/net/wifi/WifiConfiguration;
+
+    invoke-direct {p1}, Landroid/net/wifi/WifiConfiguration;-><init>()V
+
+    iput p4, p1, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
+
+    new-instance p4, Ljava/lang/StringBuilder;
+
+    const-string p6, "\""
+
+    invoke-direct {p4, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-static {p4, v0, p6}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object p4
+
+    iput-object p4, p1, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
+
+    iget-object p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mPassword:Ljava/lang/String;
+
+    if-eqz p4, :cond_5
+
+    new-instance p4, Ljava/lang/StringBuilder;
+
+    invoke-direct {p4, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mPassword:Ljava/lang/String;
+
+    invoke-static {p4, v0, p6}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object p4
+
+    iput-object p4, p1, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
+
+    iget p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
+
+    if-ne p4, p5, :cond_4
+
+    const-string p4, "connect to WPA3 access Point"
+
+    invoke-static {v1, p4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object p4, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    const/16 p5, 0x8
+
+    invoke-virtual {p4, p5}, Ljava/util/BitSet;->set(I)V
+
+    iput-boolean p3, p1, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
+
+    goto :goto_1
+
     :cond_4
-    new-instance p1, Landroid/net/wifi/WifiConfiguration;
-
-    invoke-direct {p1}, Landroid/net/wifi/WifiConfiguration;-><init>()V
-
-    iput p5, p1, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
-
-    new-instance p5, Ljava/lang/StringBuilder;
-
-    const-string v0, "\""
-
-    invoke-direct {p5, v0}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object v2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-static {p5, v2, v0}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p5
-
-    iput-object p5, p1, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
-
-    iget-object p5, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mPassword:Ljava/lang/String;
-
-    if-eqz p5, :cond_6
-
-    new-instance p5, Ljava/lang/StringBuilder;
-
-    invoke-direct {p5, v0}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object v2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mPassword:Ljava/lang/String;
-
-    invoke-static {p5, v2, v0}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p5
-
-    iput-object p5, p1, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
-
-    iget p5, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
-
-    if-ne p5, p6, :cond_5
-
-    const-string p5, "connect to WPA3 access Point"
-
-    invoke-static {v1, p5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object p5, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {p5, p2}, Ljava/util/BitSet;->set(I)V
-
-    iput-boolean p4, p1, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
+    iget-object p4, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    invoke-virtual {p4, p3}, Ljava/util/BitSet;->set(I)V
 
     goto :goto_1
 
     :cond_5
-    iget-object p2, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {p2, p4}, Ljava/util/BitSet;->set(I)V
+    iget p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
+
+    const/4 p5, 0x5
+
+    if-ne p4, p5, :cond_6
+
+    iget-object p4, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    const/16 p5, 0x9
+
+    invoke-virtual {p4, p5}, Ljava/util/BitSet;->set(I)V
+
+    iput-boolean p3, p1, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
 
     goto :goto_1
 
     :cond_6
-    iget p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSecurity:I
-
-    const/4 p5, 0x5
-
-    if-ne p2, p5, :cond_7
-
-    iget-object p2, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    const/16 p5, 0x9
-
-    invoke-virtual {p2, p5}, Ljava/util/BitSet;->set(I)V
-
-    iput-boolean p4, p1, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
-
-    goto :goto_1
+    iget-object p4, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    invoke-virtual {p4, p2}, Ljava/util/BitSet;->set(I)V
+
+    :goto_1
+    iget p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mhideSSID:I
+
+    if-ne p4, p3, :cond_7
+
+    iput-boolean p3, p1, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
 
     :cond_7
-    iget-object p2, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {p2, p3}, Ljava/util/BitSet;->set(I)V
-
-    :goto_1
-    iget p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mhideSSID:I
-
-    if-ne p2, p4, :cond_8
-
-    iput-boolean p4, p1, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
+    if-ne p4, p3, :cond_8
+
+    const-string p3, "connecting to hiddenSSID"
+
+    invoke-static {v1, p3}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+
+    iput-boolean p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mwaitingToConnect:Z
+
+    const/4 p2, -0x1
+
+    iput p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mMhsFreq:I
+
+    iget-object p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
+
+    new-instance p3, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$1;
+
+    invoke-direct {p3, p0}, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$1;-><init>(Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;)V
+
+    invoke-virtual {p2, p1, p3}, Landroid/net/wifi/WifiManager;->connect(Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+
+    return-void
 
     :cond_8
-    if-ne p2, p4, :cond_9
-
-    const-string p2, "connecting to hiddenSSID"
-
-    invoke-static {v1, p2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
     iput-boolean p3, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mwaitingToConnect:Z
 
-    const/4 p2, -0x1
-
-    iput p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mMhsFreq:I
+    const/16 p2, 0xa
+
+    iput p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->tryingToRetry:I
 
     iget-object p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
 
-    new-instance p3, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$1;
-
-    invoke-direct {p3, p0}, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$1;-><init>(Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;)V
-
-    invoke-virtual {p2, p1, p3}, Landroid/net/wifi/WifiManager;->connect(Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+    invoke-virtual {p2, p1}, Landroid/net/wifi/WifiManager;->addNetwork(Landroid/net/wifi/WifiConfiguration;)I
+
+    move-result p1
+
+    new-instance p2, Ljava/lang/StringBuilder;
+
+    const-string p4, "trying to Connect to: "
+
+    invoke-direct {p2, p4}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-virtual {p2, p4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    const-string p4, ",netId:"
+
+    invoke-virtual {p2, p4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    invoke-virtual {p2, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+
+    invoke-virtual {p2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p1
+
+    invoke-static {v1, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiScannerHandler:Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$WifiScannerHandler;
+
+    invoke-virtual {p0, p3}, Landroid/os/Handler;->sendEmptyMessage(I)Z
 
     return-void
-
-    :cond_9
-    iput-boolean p4, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mwaitingToConnect:Z
-
-    const/16 p2, 0xa
-
-    iput p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->tryingToRetry:I
-
-    iget-object p2, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiManager:Landroid/net/wifi/WifiManager;
-
-    invoke-virtual {p2, p1}, Landroid/net/wifi/WifiManager;->addNetwork(Landroid/net/wifi/WifiConfiguration;)I
-
-    move-result p1
-
-    new-instance p2, Ljava/lang/StringBuilder;
-
-    const-string p3, "trying to Connect to: "
-
-    invoke-direct {p2, p3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object p3, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-virtual {p2, p3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    const-string p3, ",netId:"
-
-    invoke-virtual {p2, p3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p2, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p1
-
-    invoke-static {v1, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner;->mWifiScannerHandler:Lcom/samsung/android/server/wifi/ap/mcf/SemWifiApContinityWifiScanner$WifiScannerHandler;
-
-    invoke-virtual {p0, p4}, Landroid/os/Handler;->sendEmptyMessage(I)Z
-
-    return-void
-.end method
+.end method
--- a/smali/com/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner.smali
+++ b/smali/com/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner.smali
@@ -803,7 +803,7 @@
 
     iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
 
-    const/16 p2, 0x8
+    const/4 p2, 0x2
 
     invoke-virtual {p1, p2}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
 
@@ -815,287 +815,270 @@
 
     iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
 
-    or-int/lit8 p1, p1, 0x4
+    or-int/2addr p1, p2
 
     iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
 
     :cond_0
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
-
-    const/4 p3, 0x2
-
-    invoke-virtual {p1, p3}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
+    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
+
+    const/4 p3, 0x1
+
+    or-int/2addr p1, p3
+
+    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
+
+    invoke-direct {p0}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->getConfiguredNetworks()Ljava/util/List;
 
     move-result-object p1
 
-    array-length p1, p1
-
-    if-eqz p1, :cond_1
-
-    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
-
-    or-int/2addr p1, p3
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
+    iget p4, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
+
+    const/4 p5, 0x3
+
+    if-ne p4, p2, :cond_1
+
+    iput p5, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
 
     :cond_1
-    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
-
-    const/4 p4, 0x1
-
-    or-int/2addr p1, p4
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->supportBand:I
-
-    invoke-direct {p0}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->getConfiguredNetworks()Ljava/util/List;
+    invoke-interface {p1}, Ljava/util/List;->iterator()Ljava/util/Iterator;
 
     move-result-object p1
 
-    iget p5, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
-
-    const/4 p6, 0x3
-
-    if-ne p5, p3, :cond_2
-
-    iput p6, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
-
     :cond_2
-    invoke-interface {p1}, Ljava/util/List;->iterator()Ljava/util/Iterator;
+    invoke-interface {p1}, Ljava/util/Iterator;->hasNext()Z
+
+    move-result p2
+
+    const/4 p4, 0x0
+
+    if-eqz p2, :cond_4
+
+    invoke-interface {p1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
+
+    move-result-object p2
+
+    check-cast p2, Landroid/net/wifi/WifiConfiguration;
+
+    iget-object p6, p2, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
+
+    if-eqz p6, :cond_2
+
+    invoke-direct {p0, p6}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->removeDoubleQuotes(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object p6
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-virtual {p6, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p6
+
+    if-eqz p6, :cond_2
+
+    iget p1, p2, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
+
+    iget-object p6, p2, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    invoke-virtual {p6, p3}, Ljava/util/BitSet;->get(I)Z
+
+    move-result p6
+
+    if-eqz p6, :cond_3
+
+    iget p6, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
+
+    if-eq p6, p5, :cond_5
+
+    if-eq p6, p3, :cond_5
+
+    :cond_3
+    iget-object p6, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
+
+    const-string v0, " existing profile removed"
+
+    invoke-static {p6, v0}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget p2, p2, Landroid/net/wifi/WifiConfiguration;->networkId:I
+
+    invoke-virtual {p0, p2}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->removeNetwork(I)V
+
+    goto :goto_0
+
+    :cond_4
+    move p1, p4
+
+    :cond_5
+    :goto_0
+    new-instance p2, Landroid/net/wifi/WifiConfiguration;
+
+    invoke-direct {p2}, Landroid/net/wifi/WifiConfiguration;-><init>()V
+
+    iput p1, p2, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
+
+    new-instance p1, Ljava/lang/StringBuilder;
+
+    const-string p6, "\""
+
+    invoke-direct {p1, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-static {p1, v0, p6}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
 
     move-result-object p1
 
-    :cond_3
-    invoke-interface {p1}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result p3
-
-    const/4 p5, 0x0
-
-    if-eqz p3, :cond_5
-
-    invoke-interface {p1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object p3
-
-    check-cast p3, Landroid/net/wifi/WifiConfiguration;
-
-    iget-object v0, p3, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
-
-    if-eqz v0, :cond_3
-
-    invoke-direct {p0, v0}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->removeDoubleQuotes(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v0
-
-    iget-object v1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-virtual {v0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_3
-
-    iget p1, p3, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
-
-    iget-object v0, p3, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {v0, p4}, Ljava/util/BitSet;->get(I)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_4
-
-    iget v0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
-
-    if-eq v0, p6, :cond_6
-
-    if-eq v0, p4, :cond_6
-
-    :cond_4
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
-
-    const-string v1, " existing profile removed"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget p3, p3, Landroid/net/wifi/WifiConfiguration;->networkId:I
-
-    invoke-virtual {p0, p3}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->removeNetwork(I)V
-
-    goto :goto_0
-
-    :cond_5
-    move p1, p5
+    iput-object p1, p2, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
+
+    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mPassword:Ljava/lang/String;
+
+    if-eqz p1, :cond_7
+
+    new-instance p1, Ljava/lang/StringBuilder;
+
+    invoke-direct {p1, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mPassword:Ljava/lang/String;
+
+    invoke-static {p1, v0, p6}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object p1
+
+    iput-object p1, p2, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
+
+    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
+
+    if-ne p1, p5, :cond_6
+
+    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
+
+    const-string p5, "connect to WPA3 access Point"
+
+    invoke-static {p1, p5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object p1, p2, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    const/16 p5, 0x8
+
+    invoke-virtual {p1, p5}, Ljava/util/BitSet;->set(I)V
+
+    iput-boolean p3, p2, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
+
+    goto :goto_1
 
     :cond_6
-    :goto_0
-    new-instance p3, Landroid/net/wifi/WifiConfiguration;
-
-    invoke-direct {p3}, Landroid/net/wifi/WifiConfiguration;-><init>()V
-
-    iput p1, p3, Landroid/net/wifi/WifiConfiguration;->meteredOverride:I
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v0, "\""
-
-    invoke-direct {p1, v0}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object v1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-static {p1, v1, v0}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p1
-
-    iput-object p1, p3, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
-
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mPassword:Ljava/lang/String;
-
-    if-eqz p1, :cond_8
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    invoke-direct {p1, v0}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object v1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mPassword:Ljava/lang/String;
-
-    invoke-static {p1, v1, v0}, Lcom/samsung/android/mcf/continuity/common/DLog$$ExternalSyntheticOutline0;->m(Ljava/lang/StringBuilder;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p1
-
-    iput-object p1, p3, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
-
+    iget-object p1, p2, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    invoke-virtual {p1, p3}, Ljava/util/BitSet;->set(I)V
+
+    goto :goto_1
+
+    :cond_7
     iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
 
-    if-ne p1, p6, :cond_7
-
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
-
-    const-string p6, "connect to WPA3 access Point"
-
-    invoke-static {p1, p6}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object p1, p3, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {p1, p2}, Ljava/util/BitSet;->set(I)V
-
-    iput-boolean p4, p3, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
+    const/4 p5, 0x5
+
+    if-ne p1, p5, :cond_8
+
+    iget-object p1, p2, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+
+    const/16 p5, 0x9
+
+    invoke-virtual {p1, p5}, Ljava/util/BitSet;->set(I)V
+
+    iput-boolean p3, p2, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
 
     goto :goto_1
 
-    :cond_7
-    iget-object p1, p3, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+    :cond_8
+    iget-object p1, p2, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
 
     invoke-virtual {p1, p4}, Ljava/util/BitSet;->set(I)V
-
-    goto :goto_1
-
-    :cond_8
-    iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSecurity:I
-
-    const/4 p2, 0x5
-
-    if-ne p1, p2, :cond_9
-
-    iget-object p1, p3, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    const/16 p2, 0x9
-
-    invoke-virtual {p1, p2}, Ljava/util/BitSet;->set(I)V
-
-    iput-boolean p4, p3, Landroid/net/wifi/WifiConfiguration;->requirePmf:Z
-
-    goto :goto_1
-
-    :cond_9
-    iget-object p1, p3, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
-
-    invoke-virtual {p1, p5}, Ljava/util/BitSet;->set(I)V
 
     :goto_1
     iget p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mhideSSID:I
 
-    if-ne p1, p4, :cond_a
-
-    iput-boolean p4, p3, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
+    if-ne p1, p3, :cond_9
+
+    iput-boolean p3, p2, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
+
+    :cond_9
+    if-ne p1, p3, :cond_a
+
+    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
+
+    const-string p3, "connecting to hiddenSSID"
+
+    invoke-static {p1, p3}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+
+    iput-boolean p4, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mwaitingToConnect:Z
+
+    const/4 p1, -0x1
+
+    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mMhsFreq:I
+
+    invoke-static {}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getInstance()Lcom/samsung/android/server/wifi/SemWifiInjector;
+
+    move-result-object p1
+
+    invoke-virtual {p1}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getWifiApThreadRunner()Lcom/samsung/android/server/wifi/ap/SemWifiApThreadRunner;
+
+    move-result-object p1
+
+    new-instance p3, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner$$ExternalSyntheticLambda0;
+
+    invoke-direct {p3, p0, p2}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner$$ExternalSyntheticLambda0;-><init>(Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;Landroid/net/wifi/WifiConfiguration;)V
+
+    invoke-virtual {p1, p3}, Lcom/samsung/android/server/wifi/ap/SemWifiApThreadRunner;->post(Ljava/lang/Runnable;)Z
+
+    return p4
 
     :cond_a
-    if-ne p1, p4, :cond_b
+    iput-boolean p3, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mwaitingToConnect:Z
+
+    const/16 p1, 0xa
+
+    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->tryingToRetry:I
 
     iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
 
-    const-string p2, "connecting to hiddenSSID"
-
-    invoke-static {p1, p2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    iput-boolean p5, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mwaitingToConnect:Z
-
-    const/4 p1, -0x1
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mMhsFreq:I
-
-    invoke-static {}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getInstance()Lcom/samsung/android/server/wifi/SemWifiInjector;
+    const-string p5, "adding Network"
+
+    invoke-static {p1, p5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    invoke-virtual {p0, p2}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->addWifiNetwork(Landroid/net/wifi/WifiConfiguration;)I
+
+    move-result p1
+
+    iget-object p2, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
+
+    new-instance p5, Ljava/lang/StringBuilder;
+
+    const-string p6, "trying to Connect to: "
+
+    invoke-direct {p5, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-object p6, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
+
+    invoke-virtual {p5, p6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    const-string p6, ",netId:"
+
+    invoke-virtual {p5, p6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    invoke-virtual {p5, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+
+    invoke-virtual {p5}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
 
     move-result-object p1
 
-    invoke-virtual {p1}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getWifiApThreadRunner()Lcom/samsung/android/server/wifi/ap/SemWifiApThreadRunner;
-
-    move-result-object p1
-
-    new-instance p2, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner$$ExternalSyntheticLambda0;
-
-    invoke-direct {p2, p0, p3}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner$$ExternalSyntheticLambda0;-><init>(Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;Landroid/net/wifi/WifiConfiguration;)V
-
-    invoke-virtual {p1, p2}, Lcom/samsung/android/server/wifi/ap/SemWifiApThreadRunner;->post(Ljava/lang/Runnable;)Z
-
-    return p5
-
-    :cond_b
-    iput-boolean p4, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mwaitingToConnect:Z
-
-    const/16 p1, 0xa
-
-    iput p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->tryingToRetry:I
-
-    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
-
-    const-string p2, "adding Network"
-
-    invoke-static {p1, p2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0, p3}, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->addWifiNetwork(Landroid/net/wifi/WifiConfiguration;)I
-
-    move-result p1
-
-    iget-object p2, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->TAG:Ljava/lang/String;
-
-    new-instance p3, Ljava/lang/StringBuilder;
-
-    const-string p6, "trying to Connect to: "
-
-    invoke-direct {p3, p6}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-object p6, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mSSID:Ljava/lang/String;
-
-    invoke-virtual {p3, p6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    const-string p6, ",netId:"
-
-    invoke-virtual {p3, p6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p3, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p3}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p1
-
     invoke-static {p2, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner;->mWifiScannerHandler:Lcom/samsung/android/server/wifi/ap/mde/SemAutohotspotWifiScanner$WifiScannerHandler;
 
-    invoke-virtual {p0, p4}, Landroid/os/Handler;->sendEmptyMessage(I)Z
-
-    return p5
+    invoke-virtual {p0, p3}, Landroid/os/Handler;->sendEmptyMessage(I)Z
+
+    return p4
 .end method
 
 .method removeNetwork(I)V
--- a/smali/com/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient$2.smali
+++ b/smali/com/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient$2.smali
@@ -2864,7 +2864,7 @@
 
     move-result-wide v16
     :try_end_5
-    .catch Ljava/lang/Exception; {:try_start_5 .. :try_end_5} :catch_5
+    .catch Ljava/lang/Exception; {:try_start_5 .. :try_end_5} :catch_4
 
     :try_start_6
     iget-object v15, v1, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient$2;->this$0:Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;
@@ -2912,82 +2912,70 @@
 
     invoke-virtual {v10, v12}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
 
+    const-string v10, "SemWifiApSmartGattClient"
+
+    const-string v12, "SEC_PRODUCT_FEATURE_WLAN_SUPPORT_80211AX_6GHZ not enabled"
+
+    invoke-static {v10, v12}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+
+    :goto_d
+    const/4 v10, 0x0
+
+    goto :goto_e
+
+    :catch_4
+    move-exception v0
+
+    const/4 v11, 0x1
+
+    goto/16 :goto_11
+
+    :cond_2b
+    const/16 v10, 0x1388
+
+    if-le v5, v10, :cond_2c
+
+    iget-object v10, v1, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient$2;->this$0:Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;
+
+    invoke-static {v10}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->-$$Nest$fgetmNl80211Manager(Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;)Landroid/net/wifi/nl80211/WifiNl80211Manager;
+
+    move-result-object v10
+
+    const/4 v12, 0x2
+
+    invoke-virtual {v10, v12}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
+
     move-result-object v10
 
     array-length v10, v10
     :try_end_7
-    .catch Ljava/lang/Exception; {:try_start_7 .. :try_end_7} :catch_5
+    .catch Ljava/lang/Exception; {:try_start_7 .. :try_end_7} :catch_4
 
     if-nez v10, :cond_2c
 
     :try_start_8
     const-string v10, "SemWifiApSmartGattClient"
 
-    const-string v12, "allowed frequency 6ghz not supported"
+    const-string v12, "allowed frequency 5ghz not supported"
 
     invoke-static {v10, v12}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
     :try_end_8
-    .catch Ljava/lang/Exception; {:try_start_8 .. :try_end_8} :catch_4
-
-    :goto_d
-    const/4 v10, 0x0
-
-    goto :goto_e
-
-    :catch_4
-    move-exception v0
-
-    const/4 v11, 0x0
-
-    goto/16 :goto_11
+    .catch Ljava/lang/Exception; {:try_start_8 .. :try_end_8} :catch_5
+
+    goto :goto_d
 
     :catch_5
     move-exception v0
 
-    const/4 v11, 0x1
+    const/4 v11, 0x0
 
     goto/16 :goto_11
-
-    :cond_2b
-    const/16 v10, 0x1388
-
-    if-le v5, v10, :cond_2c
-
-    :try_start_9
-    iget-object v10, v1, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient$2;->this$0:Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;
-
-    invoke-static {v10}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->-$$Nest$fgetmNl80211Manager(Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;)Landroid/net/wifi/nl80211/WifiNl80211Manager;
-
-    move-result-object v10
-
-    const/4 v12, 0x2
-
-    invoke-virtual {v10, v12}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
-
-    move-result-object v10
-
-    array-length v10, v10
-    :try_end_9
-    .catch Ljava/lang/Exception; {:try_start_9 .. :try_end_9} :catch_5
-
-    if-nez v10, :cond_2c
-
-    :try_start_a
-    const-string v10, "SemWifiApSmartGattClient"
-
-    const-string v12, "allowed frequency 5ghz not supported"
-
-    invoke-static {v10, v12}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-    :try_end_a
-    .catch Ljava/lang/Exception; {:try_start_a .. :try_end_a} :catch_4
-
-    goto :goto_d
 
     :cond_2c
     const/4 v10, 0x1
 
     :goto_e
-    :try_start_b
+    :try_start_9
     const-string v12, "SemWifiApSmartGattClient"
 
     new-instance v15, Ljava/lang/StringBuilder;
@@ -3067,8 +3055,8 @@
     move-result-object v5
 
     invoke-virtual {v5, v4}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
-    :try_end_b
-    .catch Ljava/lang/Exception; {:try_start_b .. :try_end_b} :catch_6
+    :try_end_9
+    .catch Ljava/lang/Exception; {:try_start_9 .. :try_end_9} :catch_6
 
     goto :goto_f
 
@@ -3083,20 +3071,20 @@
     :goto_f
     if-eqz v7, :cond_30
 
-    :try_start_c
+    :try_start_a
     const-string v4, "presharedkey"
 
     invoke-virtual {v9, v4}, Lorg/json/JSONObject;->getString(Ljava/lang/String;)Ljava/lang/String;
 
     move-result-object v15
-    :try_end_c
-    .catch Lorg/json/JSONException; {:try_start_c .. :try_end_c} :catch_7
-    .catch Ljava/lang/Exception; {:try_start_c .. :try_end_c} :catch_6
+    :try_end_a
+    .catch Lorg/json/JSONException; {:try_start_a .. :try_end_a} :catch_7
+    .catch Ljava/lang/Exception; {:try_start_a .. :try_end_a} :catch_6
 
     goto :goto_10
 
     :catch_7
-    :try_start_d
+    :try_start_b
     const-string v4, "SemWifiApSmartGattClient"
 
     const-string v5, "No Preshared key"
@@ -3209,8 +3197,8 @@
     const/4 v4, 0x0
 
     invoke-static {v4, v0, v3}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->-$$Nest$msetConnectionState(ILcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;Ljava/lang/String;)V
-    :try_end_d
-    .catch Ljava/lang/Exception; {:try_start_d .. :try_end_d} :catch_6
+    :try_end_b
+    .catch Ljava/lang/Exception; {:try_start_b .. :try_end_b} :catch_6
 
     goto :goto_13
 
@@ -3222,12 +3210,12 @@
     :catchall_1
     move-exception v0
 
-    :try_start_e
+    :try_start_c
     invoke-static/range {v16 .. v17}, Landroid/os/Binder;->restoreCallingIdentity(J)V
 
     throw v0
-    :try_end_e
-    .catch Ljava/lang/Exception; {:try_start_e .. :try_end_e} :catch_5
+    :try_end_c
+    .catch Ljava/lang/Exception; {:try_start_c .. :try_end_c} :catch_4
 
     :goto_11
     invoke-virtual {v0}, Ljava/lang/Exception;->printStackTrace()V
--- a/smali/com/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient.smali
+++ b/smali/com/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient.smali
@@ -5052,7 +5052,7 @@
 
     iput-object v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothAdapter:Landroid/bluetooth/BluetoothAdapter;
 
-    if-eqz v2, :cond_17
+    if-eqz v2, :cond_16
 
     if-nez p1, :cond_9
 
@@ -5361,9 +5361,7 @@
 
     iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
 
-    const/16 v1, 0x8
-
-    invoke-virtual {v0, v1}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
+    invoke-virtual {v0, v11}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
 
     move-result-object v0
 
@@ -5373,198 +5371,181 @@
 
     iget v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
 
-    or-int/lit8 v0, v0, 0x4
+    or-int/2addr v0, v11
 
     iput v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
 
     :cond_11
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mNl80211Manager:Landroid/net/wifi/nl80211/WifiNl80211Manager;
-
-    invoke-virtual {v0, v11}, Landroid/net/wifi/nl80211/WifiNl80211Manager;->getChannelsMhzForBand(I)[I
+    iput-boolean v10, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mNeedRetry:Z
+
+    iput v9, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mChacteristicRetryCount:I
+
+    const-string v0, "SemWifiApSmartGattClient"
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v2, "supportBand :"
+
+    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    if-eqz v8, :cond_12
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mAESKey:Ljava/lang/String;
+
+    if-nez v0, :cond_14
+
+    :cond_12
+    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
+
+    move-result v0
+
+    const/16 v1, 0xa
+
+    if-ne v0, v1, :cond_14
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothAdapter:Landroid/bluetooth/BluetoothAdapter;
+
+    if-eqz v0, :cond_13
+
+    invoke-virtual {v0}, Landroid/bluetooth/BluetoothAdapter;->getState()I
+
+    move-result v0
+
+    if-ne v0, v1, :cond_13
+
+    const-string v0, "SemWifiApSmartGattClient"
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v2, "device is not bonded, enabling BT adapter,mBluetoothIsOn:"
+
+    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-boolean v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mLocalLog:Landroid/util/LocalLog;
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v2, "SemWifiApSmartGattClient:\tdevice is not bonded, enabling BT adapter,mBluetoothIsOn:"
+
+    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    iget-boolean v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Landroid/util/LocalLog;->log(Ljava/lang/String;)V
+
+    const-string v0, "vendor.wifiap.autohotspot.btadapterenable.smartgattclient"
+
+    const-string v1, "1"
+
+    invoke-static {v0, v1}, Landroid/os/SystemProperties;->set(Ljava/lang/String;Ljava/lang/String;)V
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothAdapter:Landroid/bluetooth/BluetoothAdapter;
+
+    invoke-virtual {v0}, Landroid/bluetooth/BluetoothAdapter;->enable()Z
+
+    iput-boolean v9, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
+
+    iput-object v12, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothDevice:Landroid/bluetooth/BluetoothDevice;
+
+    goto :goto_3
+
+    :cond_13
+    const-string v0, "SemWifiApSmartGattClient"
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v2, "device is not bonded:"
+
+    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
+
+    move-result v2
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+
+    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mLocalLog:Landroid/util/LocalLog;
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v2, "SemWifiApSmartGattClient:\tdevice is not bonded:"
+
+    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
+
+    move-result v2
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Landroid/util/LocalLog;->log(Ljava/lang/String;)V
+
+    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
 
     move-result-object v0
 
-    array-length v0, v0
-
-    if-eqz v0, :cond_12
-
-    iget v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
-
-    or-int/2addr v0, v11
-
-    iput v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
-
-    :cond_12
-    iput-boolean v10, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mNeedRetry:Z
-
-    iput v9, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mChacteristicRetryCount:I
-
-    const-string v0, "SemWifiApSmartGattClient"
-
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    const-string v2, "supportBand :"
-
-    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->supportBand:I
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    if-eqz v8, :cond_13
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mAESKey:Ljava/lang/String;
+    iput-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mPendingDeviceAddress:Ljava/lang/String;
+
+    invoke-virtual {v12, v11}, Landroid/bluetooth/BluetoothDevice;->createBond(I)Z
+
+    goto :goto_3
+
+    :cond_14
+    invoke-direct {p0, v12, v10}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->tryToConnectToRemoteBLE(Landroid/bluetooth/BluetoothDevice;Z)Z
+
+    move-result v0
 
     if-nez v0, :cond_15
 
-    :cond_13
-    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
-
-    move-result v0
-
-    const/16 v1, 0xa
-
-    if-ne v0, v1, :cond_15
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothAdapter:Landroid/bluetooth/BluetoothAdapter;
-
-    if-eqz v0, :cond_14
-
-    invoke-virtual {v0}, Landroid/bluetooth/BluetoothAdapter;->getState()I
-
-    move-result v0
-
-    if-ne v0, v1, :cond_14
-
-    const-string v0, "SemWifiApSmartGattClient"
-
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    const-string v2, "device is not bonded, enabling BT adapter,mBluetoothIsOn:"
-
-    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-boolean v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mLocalLog:Landroid/util/LocalLog;
-
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    const-string v2, "SemWifiApSmartGattClient:\tdevice is not bonded, enabling BT adapter,mBluetoothIsOn:"
-
-    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    iget-boolean v2, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-virtual {v0, v1}, Landroid/util/LocalLog;->log(Ljava/lang/String;)V
-
-    const-string v0, "vendor.wifiap.autohotspot.btadapterenable.smartgattclient"
-
-    const-string v1, "1"
-
-    invoke-static {v0, v1}, Landroid/os/SystemProperties;->set(Ljava/lang/String;Ljava/lang/String;)V
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothAdapter:Landroid/bluetooth/BluetoothAdapter;
-
-    invoke-virtual {v0}, Landroid/bluetooth/BluetoothAdapter;->enable()Z
-
-    iput-boolean v9, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothIsOn:Z
-
-    iput-object v12, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mBluetoothDevice:Landroid/bluetooth/BluetoothDevice;
-
-    goto :goto_3
-
-    :cond_14
-    const-string v0, "SemWifiApSmartGattClient"
-
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    const-string v2, "device is not bonded:"
-
-    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
-
-    move-result v2
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mLocalLog:Landroid/util/LocalLog;
-
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    const-string v2, "SemWifiApSmartGattClient:\tdevice is not bonded:"
-
-    invoke-direct {v1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getBondState()I
-
-    move-result v2
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-virtual {v0, v1}, Landroid/util/LocalLog;->log(Ljava/lang/String;)V
+    const/4 v0, -0x7
+
+    invoke-direct {p0, v0, v3}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->setConnectionState(ILjava/lang/String;)V
+
+    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mFailedBLEConnections:Ljava/util/HashMap;
 
     invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
 
     move-result-object v0
 
-    iput-object v0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mPendingDeviceAddress:Ljava/lang/String;
-
-    invoke-virtual {v12, v11}, Landroid/bluetooth/BluetoothDevice;->createBond(I)Z
-
-    goto :goto_3
+    invoke-virtual {p0, v0}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
 
     :cond_15
-    invoke-direct {p0, v12, v10}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->tryToConnectToRemoteBLE(Landroid/bluetooth/BluetoothDevice;Z)Z
-
-    move-result v0
-
-    if-nez v0, :cond_16
-
-    const/4 v0, -0x7
-
-    invoke-direct {p0, v0, v3}, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->setConnectionState(ILjava/lang/String;)V
-
-    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/smarttethering/SemWifiApSmartGattClient;->mFailedBLEConnections:Ljava/util/HashMap;
-
-    invoke-virtual {v12}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-virtual {p0, v0}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
-
-    :cond_16
     :goto_3
     return v9
 
@@ -5588,7 +5569,7 @@
 
     throw p0
 
-    :cond_17
+    :cond_16
     :goto_5
     const-string p0, "SemWifiApSmartGattClient"
 
--- a/smali/com/samsung/android/server/wifi/b2b/WifiB2bPolicyManager.smali
+++ b/smali/com/samsung/android/server/wifi/b2b/WifiB2bPolicyManager.smali
@@ -223,9 +223,9 @@
 
     iput-object p1, p0, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->context:Landroid/content/Context;
 
-    const-string v0, "wifi"
-
-    invoke-virtual {p1, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
+    const-string v1, "wifi"
+
+    invoke-virtual {p1, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
 
     move-result-object p1
 
@@ -245,9 +245,16 @@
 
     iput p1, p0, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->supportedBand:I
 
+    iget p2, p0, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->setbandDefaultValue:I
+
+    if-gt p1, p2, :cond_0
+
+    iput v0, p0, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->setbandDefaultValue:I
+
+    :cond_0
     new-instance p1, Ljava/lang/StringBuilder;
 
-    const-string p2, "value of 6ghz secproduct feature :true  setbandDefaultValue "
+    const-string p2, "value of 6ghz secproduct feature :false  setbandDefaultValue "
 
     invoke-direct {p1, p2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
 
@@ -2031,31 +2038,21 @@
 .end method
 
 .method public setRoamBand(Ljava/lang/String;I)Z
-    .locals 2
+    .locals 1
 
     invoke-direct {p0}, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->getVendorVersion()I
 
+    invoke-virtual {p0, p1}, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->isPolicyApplied(Ljava/lang/String;)Z
+
     move-result v0
 
-    const/16 v1, 0xb
-
-    if-ne v0, v1, :cond_0
-
-    goto :goto_0
-
-    :cond_0
-    invoke-virtual {p0, p1}, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->isPolicyApplied(Ljava/lang/String;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_1
-
-    :goto_0
+    if-eqz v0, :cond_0
+
     const/4 p0, 0x0
 
     return p0
 
-    :cond_1
+    :cond_0
     iget-object p0, p0, Lcom/samsung/android/server/wifi/b2b/WifiB2bPolicyManager;->wifiNative:Lcom/samsung/android/server/wifi/b2b/NativeB2bRoamCommand;
 
     invoke-interface {p0, p1, p2}, Lcom/samsung/android/server/wifi/b2b/NativeB2bRoamCommand;->setRoamBand(Ljava/lang/String;I)Z
